diff --git a/CrossSolar.Tests/CrossSolar.Tests.csproj b/CrossSolar.Tests/CrossSolar.Tests.csproj
index 3801997..fc5127b 100644
--- a/CrossSolar.Tests/CrossSolar.Tests.csproj
+++ b/CrossSolar.Tests/CrossSolar.Tests.csproj
@@ -19,4 +19,8 @@
     <ProjectReference Include="..\CrossSolar\CrossSolar.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+
 </Project>
diff --git a/CrossSolar/Controllers/AnalyticsController.cs b/CrossSolar/Controllers/AnalyticsController.cs
index d2217d7..b34f7f6 100644
--- a/CrossSolar/Controllers/AnalyticsController.cs
+++ b/CrossSolar/Controllers/AnalyticsController.cs
@@ -17,14 +17,17 @@ namespace CrossSolar.Controllers
 
         private readonly IPanelRepository _panelRepository;
 
-        public AnalyticsController(IAnalyticsRepository analyticsRepository, IPanelRepository panelRepository)
+        private readonly IDayAnalyticsRepository _dayAnalyticsRepository;
+
+        public AnalyticsController(IAnalyticsRepository analyticsRepository, IPanelRepository panelRepository, IDayAnalyticsRepository dayAnalyticsRepository)
         {
             _analyticsRepository = analyticsRepository;
             _panelRepository = panelRepository;
+            _dayAnalyticsRepository = dayAnalyticsRepository;
         }
 
         // GET panel/XXXX1111YYYY2222/analytics
-        [HttpGet("{banelId}/[controller]")]
+        [HttpGet("{panelId}/[controller]")]
         public async Task<IActionResult> Get([FromRoute] string panelId)
         {
             var panel = await _panelRepository.Query()
@@ -52,7 +55,29 @@ namespace CrossSolar.Controllers
         [HttpGet("{panelId}/[controller]/day")]
         public async Task<IActionResult> DayResults([FromRoute] string panelId)
         {
-            var result = new List<OneDayElectricityModel>();
+            var panelResult = _panelRepository.Query()
+                .FirstOrDefaultAsync(x => x.Serial.Equals(panelId, StringComparison.CurrentCultureIgnoreCase));
+
+            var dateTime = DateTime.Now;
+            var analytics = await _analyticsRepository.Query()
+                .Where(x => x.PanelId.Equals(panelId, StringComparison.CurrentCultureIgnoreCase)
+                        && x.DateTime.Day == dateTime.Day
+                        && x.DateTime.Month == dateTime.Month
+                        && x.DateTime.Year == dateTime.Year).ToListAsync();
+
+            var minimum = analytics.Min(x => x.KiloWatt);
+            var maximum = analytics.Max(x => x.KiloWatt);
+            var avarage = analytics.Average(x => x.KiloWatt);
+            var sum = analytics.Sum(x => x.KiloWatt);
+
+            var result = new OneDayElectricityModel
+            {
+                DateTime = new DateTime(dateTime.Year, dateTime.Month, dateTime.Day),
+                Minimum = minimum,
+                Maximum = maximum,
+                Average = avarage,
+                Sum = sum
+            };
 
             return Ok(result);
         }
diff --git a/CrossSolar/Repository/GenericRepository.cs b/CrossSolar/Repository/GenericRepository.cs
index 0cba46f..2a76c8e 100644
--- a/CrossSolar/Repository/GenericRepository.cs
+++ b/CrossSolar/Repository/GenericRepository.cs
@@ -15,12 +15,12 @@ namespace CrossSolar.Repository
             return await _dbContext.FindAsync<T>(id);
         }
 
-        public IQueryable<T> Query()
+        public virtual IQueryable<T> Query()
         {
             return _dbContext.Set<T>().AsQueryable();
         }
 
-        public async Task InsertAsync(T entity)
+        public virtual async Task InsertAsync(T entity)
         {
             _dbContext.Set<T>().Add(entity);
             await _dbContext.SaveChangesAsync();
diff --git a/CrossSolar/Repository/IGenericRepository.cs b/CrossSolar/Repository/IGenericRepository.cs
index 09dbbce..4e46746 100644
--- a/CrossSolar/Repository/IGenericRepository.cs
+++ b/CrossSolar/Repository/IGenericRepository.cs
@@ -1,3 +1,4 @@
+using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
 
diff --git a/CrossSolar/appsettings.json b/CrossSolar/appsettings.json
index 43e8e63..6e314e0 100644
--- a/CrossSolar/appsettings.json
+++ b/CrossSolar/appsettings.json
@@ -1,6 +1,6 @@
 ﻿{
   "ConnectionStrings": {
-    "DefaultConnection": "Server=.\\SQLEXPRESS;Database=CrossSolarDb;Trusted_Connection=True;"
+    "DefaultConnection": "ServerRAJEEVP-WIN10\\SQL2016;Database=CrossSolarDb;userid=sa;password=admin@123;"
   },
   "Logging": {
     "IncludeScopes": false,
